File: Materials/Classification and Discrete Choice Models/codes/Matlab/Discrete_Choice.m
%% Discrete Choice Model with Gumbel distributed errors
%{
    Author: Jiaming Mao 
    https://jiamingmao.github.io/data-analysis/
%}

%% simulation
ccc;
n = 10000;
x = rand(n,1);
beta = [0;2] %#ok<*NOPTS>
X = [ones(n,1) x]; 
eu = [zeros(n,1) X*beta]; %expected utility; n-by-nchoice
e = randraw('extrvalue',[0 1],[n 2]); %n-by-nchoice idiosyncratic shocks
u = eu + e;
[~,y] = max(u,[],2); %m-by-n
y = y - 1;

%% estimation
b = glmfit(x,y,'binomial')

File: Materials/Classification and Discrete Choice Models/codes/R/discrete_choice.R
############################################
## Discrete Choice Model with Gumbel error #
############################################
## Code to accompany Lecture on 
## Classification and Discrete Choice Models
## Jiaming Mao (jmao@xmu.edu.cn)
## https://jiamingmao.github.io/data-analysis/

rm(list=ls())
library(VGAM) # Gumbel random number generating function

# simulate data
n = 10000
e = matrix(rgumbel(n*2),n,2) 
x = runif(n)
beta = 2
u0 = e[,1]
u1 = x*beta + e[,2] 
y = (u1 > u0)

# estimation
fit = glm(y ~ x,family="binomial")
summary(fit)




